src/plugins/widgets/topSites/TopSites.tsx

```diff
@@ -0,0 +1,45 @@
+import React, { FC, useEffect, useState } from 'react';
+import { Props, TopSitesList } from './types';
+import RectSiteBox from './RectSiteBox';
+import './TopSites.sass';
+
+export const TopSites: FC<Props> = () => {
+    const namespace: typeof chrome | typeof browser = chrome === undefined ? browser : chrome;
+    const [topSites, setTopSites] = useState<TopSitesList>([]);
+
+    // On mount, check for permissions and request if they aren't present.
+    useEffect(() => {
+        requestTopSitesPermission();
+    }, []);
+
+    if (namespace === undefined) {
+        return <h4>Your browser does not support Top Sites.</h4>
+    }
+
+    const requestTopSitesPermission = async () => {
+        try {
+            const res = await namespace.permissions.request({
+                permissions: ['topSites']
+            });
+            console.log('Successfully got permissions', res);
+        } catch (e) {
+            // TODO: Error handling? Undo widget add?
+            console.error('Failed to get TopSites permissions.', e);
+            return;
+        }
+
+        if (namespace === chrome) {
+            namespace.topSites.get(console.log);
+            namespace.topSites.get(setTopSites);
+        } else if (namespace === browser) {
+            console.log(await namespace.topSites.get());
+            setTopSites(await namespace.topSites.get());
+        } else {
+            console.error('This browser does not expose the Top Sites API.');
+        }
+    }
+
+    return <div className='TopSites-flexcontainer'>
+        {topSites.map((site) => <RectSiteBox site={site} />)}
```

Comment:
You can use `links/Display.tsx` instead. That way it matches the styling of the Quick Links widget and you don't need to add any styling yourself. It could extend it to accept optional favicons, though I am not entirely sold on including them in case the colours clash with the background. Happy to see how it looks though.